lib/widgets/content.dart

```diff
@@ -248,25 +248,245 @@ class MessageImage extends StatelessWidget {
   }
 }
 
+// Highlighted code block styles adapted from:
+// https://github.com/zulip/zulip/blob/213387249e7ba7772084411b22d8cef64b135dd0/web/styles/pygments.css
+
+// .hll { background-color: hsl(60deg 100% 90%); }
+final _kCodeBlockStyleHll = TextStyle(backgroundColor: const HSLColor.fromAHSL(1, 60, 1, 0.90).toColor());
+// .c { color: hsl(180deg 33% 37%); font-style: italic; }
+final _kCodeBlockStyleC = TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic);
+// TODO: Borders are hard in TextSpan, see the comment in `_buildInlineCode`
+// So, using a lighter background color for now (precisely it's
+// the text color used in web app in `.err` class in dark mode)
+//
+// .err { border: 1px solid hsl(0deg 100% 50%); }
+const _kCodeBlockStyleErr = TextStyle(backgroundColor: Color(0xffe2706e));
+// .k { color: hsl(332deg 70% 38%); }
+final _kCodeBlockStyleK = TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor());
+// .o { color: hsl(332deg 70% 38%); }
+final _kCodeBlockStyleO = TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor());
+// .cm { color: hsl(180deg 33% 37%); font-style: italic; }
+final _kCodeBlockStyleCm = TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic);
```

Comment:
(BTW this way of formatting these is great — I appreciate that it's so compact. No need to turn the comments into doc comments or whatever, which would inevitably make them less compact.)

As I write that, though, I realize there's one tweak which would I think improve the formatting further: add a blank line to separate each definition from the next one's comment. That just makes the grouping more unambiguous so it's quicker for the eye to go from a comment to the corresponding code and back.

It does make things a *bit* less compact, but so it goes.

That's also an advantage of moving the boilerplate to its own file — if it stayed here in this file, then making this section 50% longer would be a real cost of its own to the readability of the rest of the file, making it more of a trade-off.

Reply:
ack